pkg/operator/staticpod/controllers.go

```diff
@@ -346,14 +349,25 @@ func (b *staticPodOperatorControllerBuilder) ToControllers() (manager.Controller
 		}
 	}
 
-	manager.WithController(missingstaticpodcontroller.New(
-		b.staticPodOperatorClient,
-		b.kubeInformers.InformersFor(b.operandNamespace),
-		b.eventRecorder,
-		b.operandNamespace,
-		b.staticPodName,
-		b.operandName,
-	), 1)
+	clientConfig, err := rest.InClusterConfig()
+	if err != nil {
+		errs = append(errs, fmt.Errorf("unable to set up MissingStaticPodController: %w", err))
+	} else {
+		infra, err := clusterstatus.GetClusterInfraStatus(context.Background(), clientConfig)
```

Comment:
I was using a simple client because the fields I am looking for inside the `infrastructure` are immutable, hence the simplicity of passing the actual value once it has been read. This resource is not going to be updated/mutated once the cluster is installed.
By using a lister I understand you mean checking the actual value that I am after in every `sync` for the `missingStaticPodController`, right? I was trying to avoid extra calls to the API server, but not sure how often the `sync` is called, tbh.

Reply:
I was referring to `configlistersv1.InfrastructureLister`. In general a lister caches data from the API, so no additional calls to the server on every `sync`.

Reply:
please have a look at https://github.com/openshift/cluster-kube-apiserver-operator/blob/master/pkg/operator/startupmonitorreadiness/enablement.go#L16

Reply:
I would like to avoid calling the API server during startup, it can fail and take down the entire operator.

Reply:
My initial thought was along these lines, so let me get it back to the original implementation :)
Thanks!